libsolidity/ast/Types.cpp

```diff
@@ -1155,6 +1157,10 @@ FixedPointType const* RationalNumberType::fixedPointType() const
 		fractionalDigits++;
 	}
 
+	// We need more than 80 digits of precision.
+	if (fractionalDigits >= 80 && value.denominator() != 1)
```

Comment:
```suggestion
	if (fractionalDigits >= 80 || value.denominator() != 1)
```
?

Reply:
I think I meant it the way I wrote it: The loop tries to find a precision so that the scaled value is non-fractional. It tries until either the value is too large to fit the number of bits or until it needs more than 80 fractional digits. The condition here should trigger if it exitend the loop because it needed more than 80 digits and the scaled value is still fractional.